mozci/push.py

```diff
@@ -375,6 +375,82 @@ def scheduled_duration(self):
 
         return int(duration / 3600)
 
+    def _check_classification(self, failure_push, classifications):
+        """Checks a 'fixed by commit' classification to figure out what push it references.
+
+        Returns:
+            bool or None: True, if the classification references this push.
+                          False, if the classification references another push.
+                          None, if it is not clear what the classification references.
+        """
+        fixed_by_commit_classification_notes = [
+            n[:12]
+            for c, n in classifications
+            if c == "fixed by commit"
+            if n is not None
+        ]
+
+        if len(fixed_by_commit_classification_notes) == 0:
+            return None
+
+        # If the failure was classified as fixed by commit, and the fixing commit
+        # is a backout of the current push, it is definitely a regression of the
+        # current push.
+        if (
+            self.backedout
+            and self.backedoutby[:12] in fixed_by_commit_classification_notes
+        ):
+            return True
+
+        # If the failure was classified as fixed by commit, and the fixing commit
+        # is a backout of another push, it is definitely not a regression of the
+        # current push.
+        # Unless some condition holds which makes us doubt about the correctness of the
+        # classification.
+        # - the backout commit also backs out one of the commits of this push;
+        # - the other push backed-out by the commit which is mentioned in the classification
+        #   is landed after the push where the failure occurs (so, it can't have caused it);
+        # - the backout push also contains a commit backing out one of the commits of this push.
+        for classification_note in fixed_by_commit_classification_notes:
+            fix_hgmo = HGMO.create(classification_note, branch=self.branch)
```

Comment:
We should probably catch `PushNotFound` and return `None` here in case there is a typo in the revision.

Reply:
Good idea, and I've seen this a few times on a few past pushes. Fixed with ff4c978 (I'm still printing a warning).